Source Pane visibility behavior

melissa-barca opened the pull request:
Fix bug that prevented ensuring the Source Pane was visible when a user opened a new file and ensured a new file was opened every time the Source Pane attempted to open without a file.

Fixes most of #7145 but I haven't been able to reproduce the behavior where a new untitled file is opened when an existing document is opened and there are no other files open. 

Also fixes issue where commands weren't being handled from the SourceColumnManager.

jmcphers commented on src/gwt/src/org/rstudio/studio/client/workbench/views/source/SourceColumn.java:
```diff
@@ -939,9 +940,9 @@ private void manageCommands()
    public void newDoc(EditableFileType fileType,
                       ResultCallback<EditingTarget, ServerError> callback)
    {
-      ensureVisible(true);
```
Don't we eventually need to call this so that e.g. the source pane opens for non-file tab types? (e.g. `View(mtcars)`)

melissa-barca replied:
Yes, thank you. I had intended to change this back before submitting the PR.

melissa-barca commented:
I updated the code to properly handle all of the cases including the new ones regarding when source files are opened. I also added a fix that was preventing commands from executing from the column manager.
